Jenkinsfile

```diff
@@ -160,14 +160,14 @@ pipeline {
                         deleteDir()
                         unstash 'MathSetup'
                         sh "echo CXX=${MPICXX} >> make/local"
-                        sh "echo CXX_TYPE=gcc >> make/local"
+                        sh "echo CXX_TYPE=gcc >> make/local"                        
                         sh "echo STAN_MPI=true >> make/local"
                         runTests("test/unit")
                     }
                     post { always { retry(3) { deleteDir() } } }
                 }
                 stage('Full unit with GPU') {
-                    agent { label "gpu" }
+                    agent { label "gpu && linux" }
```

Comment:
Do you intend to leave these changes to jenkinsfile in? These will probably make jenkins detect more similar bugs with async calls.

Reply:
I forgot to revert this back. I think it would be best to not restrict this as this machine would become a bottleneck for testing. 

We have another Windows + NVIDIA machine that just needs to be setup for Jenkins that will be also used for these tests. Once we have that we can maybe think about running it on NVIDIA only.